plugins/lua/tiletypes.lua

```diff
@@ -0,0 +1,29 @@
+local _ENV = mkmodule('plugins.tiletypes')
+
+local utils = require('utils')
+
+local function parse_cursor(opts, arg)
+    local _, _, x, y, z = arg:find('^%s*(%d+)%s*,%s*(%d+)%s*,%s*(%d+)%s*$')
+    if not x then
+        qerror(('invalid argument for --cursor option: "%s"; expected format' ..
+                ' is "<x>,<y>,<z>", for example: "30,60,150"'):format(arg))
+    end
+    opts.cursor.x = tonumber(x)
+    opts.cursor.y = tonumber(y)
+    opts.cursor.z = tonumber(z)
+end
+
+function parse_commandline(opts, ...)
```

Comment:
Would it be worth trying to pull in a C++ argument-parsing module to avoid plugins delegating it to Lua? My main concern is complexity - overhead is probably insignificant. (Counterpoint: there are some benefits to having a consistent parser across tools, regardless of the language they're written in.)

Reply:
Is delegating arg parsing to Lua a bad thing? String parsing isn't exactly the strong suit of C++. In my experience, it's very verbose and very error-prone. There is toil with the `_identity` class and funky-looking code that calls Lua, true, and I can see how that could be off-putting, but the power and flexibility of Lua for arg parsing is a powerful draw. Since this use case is fairly well-defined, perhaps we could wrap it in a friendlier API?

Reply:
Yeah, avoiding complexity is a good point, so that's another argument in favor besides the consistency-across-languages argument. If we want plugins to be able to easily modify their own custom structures from Lua, that led to my other comment at https://github.com/DFHack/dfhack/pull/1867#discussion_r661124795. This is fine now if we're willing to put in the manual effort to maintain the field definitions for those structures (just a struct of arguments in this case).